internal/formats/report_writer.go

```diff
@@ -0,0 +1,144 @@
+package formats
```

Comment:
I wonder if this would be better in `syft/format`? And/or possibly abstracted with an interface so Grype could also use this code to support multiple outputs in a similar way?